Temporarily disable test that check const fields

mihails-strasuns-sociomantic opened the pull request:
Because of deficiency in DMD implementation, combination of `pragma(msg)` and
`is(typeof())` will result in message printed even if instantiation is
speculative. As it is not possible to turn them into asserts yet, all tests that
trigger warning are disabled for now.

mihails-strasuns-sociomantic commented:
(we should release 2.6.6 with this ASAP to remove irrelevant warning spam in 2.6.5)

gavin-norman-sociomantic commented on src/ocean/util/serialize/contiguous/package_.d:
```diff
@@ -734,10 +734,10 @@ unittest
     static struct CS
     {
-        cstring s;
+        mstring s;
```
Will this change need to be reverted in v4.x.x as well?

mihails-strasuns-sociomantic replied:
No, this will be a breaking change in `v4.x.x` - such struct will no longer deserialize (it was never intended to). I wonder though how will it affect bidding apps as I presume this test was added by @mathias-lang-sociomantic originally.

mihails-strasuns-sociomantic replied:
Ok, we talked with @mathias-lang-sociomantic, looks like it was needed for one of work-in-progress ports and doesn't affect any merged code. We'll see if that code can be adjusted in a fashion simlar to swarm, for now this adjustments should be OK to merge.

gavin-norman-sociomantic replied:
Is this change unrelated to "Temporarily disable test that check const fields" then?

mihails-strasuns-sociomantic replied:
A bit indirectly. I will split it into separate commit and explain there.

mihails-strasuns-sociomantic replied:
Done

gavin-norman-sociomantic commented on src/ocean/util/serialize/contiguous/package_.d:
```diff
@@ -782,10 +782,14 @@ version (D_Version2) unittest
-                  "Serializer should reject a struct with 'istring'");
-    static assert(!is(typeof({Deserializer.deserialize!(II)(buffer2);})),
-                  "Deserializer should reject a struct with 'immutable' element");
+    version (none)
```
We should create an issue in v4.0.0 to reenable these tests.

mihails-strasuns-sociomantic replied:
https://github.com/sociomantic-tsunami/ocean/issues/95

mihails-strasuns-sociomantic committed:
Temporarily disable test that check const fields

Because of deficiency in DMD implementation, combination of `pragma(msg)` and
`is(typeof())` will result in message printed even if instantiation is
speculative. As it is not possible to turn them into asserts yet, all tests that
trigger warning are disabled for now.

mihails-strasuns-sociomantic committed:
Modify test that was doing const deserialization

It slipped through review by an accident some time ago - such use case was not
intended and is going to be completely prohibited in ocean v4.0.0

Instead applications are supposed to use mutable structs for deserialization and
apply head `const` if needed on serialized version.

gavin-norman-sociomantic commented:
LGTM.
